backend/src/db/providers/userProvider.ts

```diff
@@ -1,7 +1,24 @@
 import UserModel from 'db/models/User';
 
 const getUserProvider = async (userId: string) => {
-  const dbUser = await UserModel.findById(userId).lean();
+  const dbUser = await UserModel.findById(userId)
+    .populate({
+      path: 'pendingCourses',
+      select: 'user course status',
+      populate: [
+        {
+          path: 'course',
+          model: 'Courses',
+          select: 'title',
+        },
+        {
+          path: 'user',
+          model: 'User',
+          select: 'firstName lastName position avatar',
+        },
+      ],
+    })
```

Comment:
we should not  populate our user here, just stay with array of pending-courses id
you can create new  provider and call it for example "getPendingCourseProvider" - to get only pending course

Reply:
done